libr/core/visual.c

```diff
@@ -845,13 +845,16 @@ static void reset_print_cur(RPrint *p) {
 static void visual_offset(RCore *core) {
 	char buf[256];
 	r_line_set_prompt ("[offset]> ");
+	core->cons->line->offset_prompt = true;
+	core->cons->line->offset_index = 0;
```

Comment:
what about doing: r_line_set_history_callback()  and implement the logic of offset_prompt in here?